src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.cs

```diff
@@ -934,17 +991,33 @@ private void UpdateHyperlinks()
 			{
 				InsertHandler(PointerPressedEvent, OnPointerPressed);
 				InsertHandler(PointerReleasedEvent, OnPointerReleased);
+				InsertHandler(PointerMovedEvent, OnPointerMoved);
+				InsertHandler(PointerEnteredEvent, OnPointerEntered);
+				InsertHandler(PointerExitedEvent, OnPointerExit);
 				InsertHandler(PointerCaptureLostEvent, OnPointerCaptureLost);
 			}
 			else if (!HasHyperlink && previousHasHyperlinks)
 			{
 				RemoveHandler(PointerPressedEvent, OnPointerPressed);
 				RemoveHandler(PointerReleasedEvent, OnPointerReleased);
+				RemoveHandler(PointerMovedEvent, OnPointerMoved);
+				RemoveHandler(PointerEnteredEvent, OnPointerEntered);
+				RemoveHandler(PointerExitedEvent, OnPointerExit);
 				RemoveHandler(PointerCaptureLostEvent, OnPointerCaptureLost);
 			}
 		}
 
-		private bool HasHyperlink => _hyperlinks.Any();
+		private bool HasHyperlink
+		{
+			get
+			{
+				var hasHyperlink = _hyperlinks.Any();
```

Comment:
Since we are modifying these lines, let's use the more efficient Count property :)
```suggestion
				var hasHyperlink = _hyperlinks.Count;
```